src/Common/src/System/Net/Security/Unix/SafeFreeNegoCredentials.cs

```diff
@@ -59,10 +59,7 @@ public SafeFreeNegoCredentials(bool isNtlmOnly, string username, string password
                 domain = domain.Trim();
             }
 
-            if (username != null)
-            {
-                username = username.Trim();
-            }
+            username = username.Trim();
```

Comment:
cc: @bartonjs
We assert username is not null earlier, and then we dereference it earlier, so there's no use in doing a null check here.